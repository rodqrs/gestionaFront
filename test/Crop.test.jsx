import { test, describe, beforeAll, beforeEach, vi, expect } from 'vitest';
import { render } from '@testing-library/react';
import Crop from '../src/views/Crop';
import { WrapperTest } from './WrapperTest';
import * as cropServices from '../src/services/cropService';

vi.mock('../src/services/cropService', () => ({
 fetchCrops: vi.fn(),
}));

describe('Crop View', () => {
 beforeAll(() => {
  //polyfill para simular el sessionStorage
  const storageMock = (() => {
   let store = {}; // Simula el almacenamiento
   return {
    getItem: (key) => store[key] || null,
    setItem: (key, value) => {
     store[key] = String(value);
    },
    removeItem: (key) => {
     delete store[key];
    },
    clear: () => {
     store = {};
    },
   };
  })();

  Object.defineProperty(global, 'sessionStorage', {
   value: storageMock,
  });
 });
 beforeEach(() => {
  const mockUserData = JSON.stringify({
   id: 123,
   nombre: 'Juan Pérez',
   email: 'd@d.com',
   projectsByUser: [
    {
     id: 1,
     nombre: 'Proyecto 1',
     crops: [{ id: 1, nombre: 'Maíz', season: [] }],
    },
   ],
  });

  vi.spyOn(sessionStorage, 'getItem').mockImplementation((key) => {
   if (key === 'user_data') {
    return mockUserData;
   }
   return null;
  });
 });

 test('should render  Crop View', async () => {
  const mockCrops = [
   {
    id: 1,
    nombre: 'Maíz',
    tipo_siembra: 'Primavera',
    fecha_inicio: '2023-01-01',
    area_terreno: 100,
    unidad: 'hectárea',
    descripcion: 'Unidad de medida para área',
   },
   {
    id: 2,
    nombre: 'Trigo',
    tipo_siembra: 'Verano',
    fecha_inicio: '2023-02-01',
    area_terreno: 200,
    unidad: 'hectárea',
    descripcion: 'Unidad de medida para área',
   },
  ];

  cropServices.fetchCrops.mockResolvedValue(mockCrops);

  const { container } = render(
   <WrapperTest>
    <Crop />
   </WrapperTest>
  );

  expect(container).toMatchSnapshot();
  expect(container.querySelector('.crop__home-view')).toBeTruthy();
 });
});
