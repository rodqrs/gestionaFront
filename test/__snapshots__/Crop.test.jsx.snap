// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Crop View > should render  Crop View 1`] = `
<div>
  <div
    class="crop__home-view"
  >
    <aside
      class="sidebar"
    >
      <header
        class="sidebar__header"
      >
        <a
          class=""
          data-discover="true"
          href="/home"
        >
          <svg
            fill="#008164"
            height="60"
            viewBox="0 0 14 16"
            width="52"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.792 5.246a4.008 4.008 0 0 1-.881.848 2.347 2.347 0 0 0 .465-1.35 4.676 4.676 0 0 0-.437-.404 12.463 12.463 0 0 1-2.123 5.273c-.253.356-.512.682-.772.98a2.094 2.094 0 0 1 1.274-.31 2.115 2.115 0 0 1 1.33.64c-.117.142-.788.926-1.916.993a2.686 2.686 0 0 1-1.573-.41c-.34.317-.668.586-.967.813a4.692 4.692 0 0 0 5.6-7.072Zm-1.091 1.56a2.104 2.104 0 0 1 1.09-.544c.014.086.106.706-.336 1.214-.528.608-1.325.49-1.371.482A2.11 2.11 0 0 1 9.7 6.806Zm.293 2.932c-.758.406-1.493.031-1.55 0 .039-.122.263-.775.963-1.13a1.969 1.969 0 0 1 1.443-.123c-.009.073-.118.859-.856 1.253Z"
            />
            <path
              d="M3.792 9.696c.044-.987.546-1.66.734-1.888.23.38.945.922 1.026 2.14a3.24 3.24 0 0 1-.476 1.9 11.05 11.05 0 0 0 1.817-1.691c.481-.561 1.368-1.616 1.928-3.267a9.237 9.237 0 0 0 .485-2.984 4.72 4.72 0 0 0-.613-.295 4.574 4.574 0 0 1-.04 2.316 4.028 4.028 0 0 0-.884-2.572 4.692 4.692 0 0 0-3.043 8.741c-.152-.156-.993-1.05-.933-2.4Zm3.101-5.361c.252.21.868.394 1.257 1.089.534.953.084 1.969.084 2.178-.129-.044-.958-.48-1.34-1.424a2.534 2.534 0 0 1 0-1.843Zm-1.215 3.56c.088-.484.294-.916.46-1.13.61.553.704.607.88 1.34.276 1.138-.517 2.02-.586 2.094-.419-.922-.977-1.076-.754-2.304Z"
            />
            <path
              d="M6.953 0 0 3.984v8.02L6.953 16l6.869-4.063V3.853L6.952 0Zm5.825 11.328-5.831 3.449-5.903-3.392V4.576l5.903-3.382 5.831 3.271v6.863Z"
            />
          </svg>
          <div
            class="header__slogan"
          >
            <h1
              class="slogan__title"
            >
              GestionA
            </h1>
            <h2
              class="slogan__description"
            >
              Sistema de gestión agrícola
            </h2>
          </div>
        </a>
        <svg
          fill="none"
          height="20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M14.592 6.69a.72.72 0 0 1 0 1.02L10.3 12l4.29 4.29a.72.72 0 1 1-1.018 1.02l-4.8-4.8a.72.72 0 0 1 0-1.02l4.8-4.8a.72.72 0 0 1 1.019 0Z"
            fill="#008164"
            fill-rule="evenodd"
          />
        </svg>
      </header>
      <nav
        class="sidebar__overwiew"
      >
        <p
          class="sidebar__label"
        >
          Gestión de Cultivos
        </p>
        <ul
          class="sidebar__list"
        >
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/crop"
            >
              <svg
                fill="none"
                viewBox="0 0 17 17"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.241 16.5V8.224m0 .517v3.104m0-3.104a4.138 4.138 0 0 0-4.138-4.138H1v3.104a4.138 4.138 0 0 0 4.138 4.138H8.24m0-3.104h3.104a4.138 4.138 0 0 0 4.138-4.138V1.5h-3.104a4.138 4.138 0 0 0-4.138 4.138V8.74Zm0 0 4.138-4.138m-4.138 7.242L4.103 7.707"
                  stroke="#008164"
                  stroke-width="1.2"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Cultivo
              </span>
            </a>
          </li>
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/season"
            >
              <svg
                fill="none"
                viewBox="0 0 16 12"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M15.483 2.218A1.609 1.609 0 0 0 13.876.61H2.09A1.607 1.607 0 0 0 .483 2.218v7.565A1.607 1.607 0 0 0 2.09 11.39h11.786a1.609 1.609 0 0 0 1.607-1.607V2.218Zm-1.072 0v7.565a.535.535 0 0 1-.535.535H2.09a.536.536 0 0 1-.536-.535V2.218a.534.534 0 0 1 .536-.536h11.786a.534.534 0 0 1 .535.536Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="m9.59 4.4 2.145.006c.295 0 .534.24.534.536v2.675a.535.535 0 0 1-.536.536h-7.5a.537.537 0 0 1-.536-.536V4.94c0-.295.239-.535.533-.536l1.61-.007a.536.536 0 0 0-.004-1.072l-1.61.008c-.885.003-1.6.722-1.6 1.607v2.677a1.607 1.607 0 0 0 1.607 1.607h7.5a1.605 1.605 0 0 0 1.607-1.607V4.942c0-.886-.716-1.605-1.602-1.607L9.59 3.329a.537.537 0 0 0-.775-.47l-1.071.535a.535.535 0 0 0 0 .958l1.071.536a.536.536 0 0 0 .775-.48v-.007Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Temporada
              </span>
            </a>
          </li>
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/activities"
            >
              <svg
                fill="none"
                viewBox="0 0 15 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4.5 1.5A.5.5 0 0 1 5 2v.5h5V2a.5.5 0 0 1 1 0v.5h1.5A1.5 1.5 0 0 1 14 4v9a1.5 1.5 0 0 1-1.5 1.5h-10A1.5 1.5 0 0 1 1 13V4a1.5 1.5 0 0 1 1.5-1.5H4V2a.5.5 0 0 1 .5-.5Zm5.5 2V4a.5.5 0 0 0 1 0v-.5h1.5a.5.5 0 0 1 .5.5v1.5H2V4a.5.5 0 0 1 .5-.5H4V4a.5.5 0 0 0 1 0v-.5h5Zm-8 3V13a.5.5 0 0 0 .5.5h10a.5.5 0 0 0 .5-.5V6.5H2ZM7 8a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm2.5-.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1ZM11 8a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm.5 1.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1ZM9 10a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm-1.5-.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1ZM5 10a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm-1.5-.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1ZM3 12a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm2.5-.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1ZM7 12a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm2.5-.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Actividades
              </span>
            </a>
          </li>
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/supply"
            >
              <svg
                fill="none"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M15 8c0 .5-.5.5-.5.5H13V10s0 .5-.5.5-.5-.5-.5-.5V8.5h-1.5s-.5 0-.5-.5.5-.5.5-.5H12V6s0-.5.5-.5.5.5.5.5v1.5h1.5s.5 0 .5.5ZM3.483 6.5H6l.017-4H3.5l-.017 4Zm-1.5 7H6v-6H1.983v6ZM12.5 12c-.5 0-.5.5-.5.5v1H7V5h2.483S10 5 10 4.5 9.5 4 9.483 4H7V2s0-.5-.517-.5H3s-.5 0-.5.5v4.5h-1S1 6.5 1 7v7c0 .5.5.5.5.5h10.983c.412 0 .517-.088.517-.5v-1.5s0-.5-.5-.5Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Inventario
              </span>
            </a>
          </li>
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/sales"
            >
              <svg
                fill="none"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#a)"
                >
                  <path
                    d="M7.045 13.39c0-.129.105-.234.235-.234h2.343a.235.235 0 0 1 0 .469H7.28a.234.234 0 0 1-.235-.234Zm0-3.984a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm1.407 0a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm1.406 0a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm0-1.406a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938ZM8.452 8a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938ZM7.045 8a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm0-1.406a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm1.407 0a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm1.406 0a.469.469 0 1 0 0 .938.469.469 0 0 0 0-.938Zm-.703-4.219H7.748a.234.234 0 0 0 0 .469h1.407a.234.234 0 0 0 0-.469Zm0 .938H7.748a.234.234 0 0 0 0 .468h1.407a.234.234 0 0 0 0-.469Zm4.453 8.437v1.875c0 .777-.63 1.406-1.407 1.406h-7.5c-.776 0-1.406-.63-1.406-1.406V11.75l.792-6.727c.083-.708.683-1.33 1.396-1.33h.976V1.437c0-.258.328-.468.586-.468h2.813c.258 0 .586.21.586.469v2.255h.976c.713 0 1.313.622 1.396 1.33l.792 6.727ZM7.045 4.719h2.813V1.437H7.045V4.72Zm-3.18 7.031H13l-.785-6.672c-.056-.472-.32-.828-.795-.828h-.976v.469h.241S11 4.729 11 5s-.315.271-.315.271H6.342S6 5.271 6 5s.342-.281.342-.281h.117V4.25h-.976c-.475 0-.777.356-.833.828l-.785 6.672Zm9.135.469H3.865v1.406c0 .517.32.84.837.84h7.5c.517 0 .798-.323.798-.84v-1.406Z"
                    fill="#008164"
                  />
                </g>
                <defs>
                  <clippath
                    id="a"
                  >
                    <path
                      d="M.483.5h15v15h-15z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
              <span
                class="option__link--text"
              >
                Ventas
              </span>
            </a>
          </li>
        </ul>
      </nav>
      <nav
        class="sidebar__overwiew"
      >
        <p
          class="sidebar__label"
        >
          Métricas
        </p>
        <ul
          class="sidebar__list"
        >
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/information"
            >
              <svg
                fill="none"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#a)"
                >
                  <path
                    d="m14.817 13.69-1.59-1.59a3.428 3.428 0 0 0-.306-4.52 3.453 3.453 0 0 0-2.438-1.011 3.453 3.453 0 0 0-3.447 3.447 3.453 3.453 0 0 0 3.449 3.45c.783 0 1.504-.265 2.082-.707l1.591 1.59a.46.46 0 0 0 .657 0 .465.465 0 0 0 .002-.659Zm-6.85-3.674a2.521 2.521 0 0 1 2.518-2.518 2.518 2.518 0 1 1-2.518 2.518Zm3.959-5.544a.465.465 0 0 1-.466.465H6.801a.465.465 0 0 1 0-.93h4.658c.257 0 .466.208.466.465Zm-8.928 9.082h3.105a.467.467 0 0 1 0 .932H2.998a1.556 1.556 0 0 1-1.553-1.553V4.646c0-.209.075-.411.209-.572l2.35-2.796a.841.841 0 0 1 .643-.3h7.588c.856 0 1.554.696 1.554 1.553v3.337a.466.466 0 1 1-.932 0V2.53a.621.621 0 0 0-.62-.62H5.405v2.694a.743.743 0 0 1-.747.734h-2.28v7.596c0 .342.279.62.62.62ZM4.474 2.167 2.59 4.407h1.883v-2.24Zm1.164 6.975a.466.466 0 1 0 0-.932h-1.63a.466.466 0 1 0 0 .932h1.63Zm0 2.14a.466.466 0 1 0 0-.931h-1.63a.466.466 0 0 0 0 .93h1.63Zm.465-5.174H4.008a.466.466 0 1 0 0 .93h2.095a.466.466 0 0 0 0-.93Z"
                    fill="#008164"
                  />
                </g>
                <defs>
                  <clippath
                    id="a"
                  >
                    <path
                      d="M.483.5h15v15h-15z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
              <span
                class="option__link--text"
              >
                Información
              </span>
            </a>
          </li>
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/dashboard"
            >
              <svg
                fill="none"
                viewBox="0 0 15 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M2.8 1.5h-.05c-.229 0-.426 0-.6.042A1.5 1.5 0 0 0 1.04 2.65C1 2.824 1 3.02 1 3.25V5.75c0 .229 0 .426.041.6A1.5 1.5 0 0 0 2.15 7.46c.174.042.371.041.6.041H5.25c.229 0 .426 0 .6-.041A1.5 1.5 0 0 0 6.96 6.35C7 6.176 7 5.98 7 5.75V3.25c0-.229 0-.426-.041-.6A1.5 1.5 0 0 0 5.85 1.542C5.676 1.5 5.48 1.5 5.25 1.5H2.8Zm-.417 1.014c.043-.01.11-.014.417-.014h2.4c.308 0 .374.004.417.014a.5.5 0 0 1 .37.37c.01.042.013.108.013.416v2.4c0 .308-.003.374-.014.417a.5.5 0 0 1-.37.37c-.042.01-.108.013-.416.013H2.8c-.308 0-.374-.003-.417-.014a.5.5 0 0 1-.37-.37C2.004 6.075 2 6.009 2 5.7V3.3c0-.308.003-.374.014-.417a.5.5 0 0 1 .37-.37ZM9.8 1.5h-.05c-.229 0-.426 0-.6.042A1.5 1.5 0 0 0 8.04 2.65C8 2.824 8 3.02 8 3.25V5.75c0 .229 0 .426.041.6A1.5 1.5 0 0 0 9.15 7.46c.174.042.371.041.6.041H12.25c.229 0 .426 0 .6-.041A1.5 1.5 0 0 0 13.96 6.35C14 6.176 14 5.98 14 5.75V3.25c0-.229 0-.426-.041-.6a1.5 1.5 0 0 0-1.109-1.108c-.174-.042-.371-.042-.6-.042H9.8Zm-.417 1.014c.043-.01.11-.014.417-.014h2.4c.308 0 .374.004.417.014a.5.5 0 0 1 .37.37c.01.042.013.108.013.416v2.4c0 .308-.004.374-.014.417a.5.5 0 0 1-.37.37c-.042.01-.108.013-.416.013H9.8c-.308 0-.374-.003-.417-.014a.5.5 0 0 1-.37-.37C9.004 6.075 9 6.009 9 5.7V3.3c0-.308.003-.374.014-.417a.5.5 0 0 1 .37-.37ZM2.75 8.5H5.25c.229 0 .426 0 .6.042A1.5 1.5 0 0 1 6.96 9.65c.041.174.041.371.041.6V12.75c0 .229 0 .426-.041.6A1.5 1.5 0 0 1 5.85 14.46c-.174.042-.371.041-.6.041H2.75c-.229 0-.426 0-.6-.041A1.5 1.5 0 0 1 1.04 13.35C1 13.176 1 12.98 1 12.75V10.25c0-.229 0-.426.041-.6A1.5 1.5 0 0 1 2.15 8.542c.174-.042.371-.042.6-.042Zm.05 1c-.308 0-.374.004-.417.014a.5.5 0 0 0-.37.37c-.01.042-.013.108-.013.416v2.4c0 .308.003.374.014.417a.5.5 0 0 0 .37.37c.042.01.108.013.416.013h2.4c.308 0 .374-.003.417-.014a.5.5 0 0 0 .37-.37c.01-.042.013-.108.013-.416v-2.4c0-.308-.003-.374-.014-.417a.5.5 0 0 0-.37-.37c-.042-.01-.108-.013-.416-.013H2.8Zm7-1h-.05c-.229 0-.426 0-.6.042A1.5 1.5 0 0 0 8.04 9.65C8 9.824 8 10.02 8 10.25V12.75c0 .229 0 .426.041.6A1.5 1.5 0 0 0 9.15 14.46c.174.042.371.041.6.041H12.25c.229 0 .426 0 .6-.041a1.5 1.5 0 0 0 1.109-1.109c.041-.174.041-.371.041-.6V10.25c0-.229 0-.426-.041-.6a1.5 1.5 0 0 0-1.109-1.108c-.174-.042-.371-.042-.6-.042H9.8Zm-.417 1.014c.043-.01.11-.014.417-.014h2.4c.308 0 .374.004.417.014a.5.5 0 0 1 .37.37c.01.042.013.108.013.416v2.4c0 .308-.004.374-.014.417a.5.5 0 0 1-.37.37c-.042.01-.108.013-.416.013H9.8c-.308 0-.374-.003-.417-.014a.5.5 0 0 1-.37-.37C9.004 13.075 9 13.009 9 12.7v-2.4c0-.308.003-.374.014-.417a.5.5 0 0 1 .37-.37Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Dashboard
              </span>
            </a>
          </li>
        </ul>
      </nav>
      <nav
        class="sidebar__account"
      >
        <p
          class="sidebar__label"
        >
          Cuenta
        </p>
        <ul
          class="sidebar__list"
        >
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/profile"
            >
              <svg
                fill="none"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.554 1.15a.85.85 0 0 0-.83.662l-.237 1.05c-.38.11-.74.262-1.08.448l-.91-.574a.85.85 0 0 0-1.055.118l-.606.606a.85.85 0 0 0-.118 1.054l.574.912c-.186.338-.337.7-.447 1.079l-1.05.238a.85.85 0 0 0-.662.829v.857a.85.85 0 0 0 .662.829l1.05.238c.11.379.261.74.448 1.08l-.575.91a.85.85 0 0 0 .118 1.055l.606.606a.85.85 0 0 0 1.055.118l.91-.574c.34.186.7.337 1.08.447l.238 1.05a.85.85 0 0 0 .829.662h.857a.85.85 0 0 0 .829-.662l.238-1.05c.379-.11.74-.26 1.08-.447l.91.574a.85.85 0 0 0 1.055-.118l.606-.606a.85.85 0 0 0 .118-1.054l-.574-.911c.187-.34.338-.7.448-1.08l1.05-.238a.85.85 0 0 0 .662-.83v-.856a.85.85 0 0 0-.662-.83l-1.05-.237c-.11-.38-.261-.74-.448-1.08l.575-.911a.85.85 0 0 0-.118-1.054l-.607-.606a.85.85 0 0 0-1.054-.118l-.91.574a5.321 5.321 0 0 0-1.081-.448l-.238-1.05a.85.85 0 0 0-.83-.662h-.856Zm-2.15 3.162a4.476 4.476 0 0 1 1.794-.744L7.554 2h.857l.355 1.568c.66.116 1.27.375 1.796.744l1.36-.857.607.606-.858 1.36c.37.527.628 1.136.744 1.795l1.568.356v.857l-1.568.355a4.476 4.476 0 0 1-.744 1.796l.857 1.36-.606.606-1.36-.857a4.476 4.476 0 0 1-1.796.743L8.411 14h-.857l-.356-1.568a4.475 4.475 0 0 1-1.794-.744l-1.36.858-.607-.606.858-1.36a4.476 4.476 0 0 1-.744-1.796L1.983 8.43v-.857l1.568-.356a4.475 4.475 0 0 1 .743-1.794L3.437 4.06l.606-.606 1.36.857ZM9.507 8a1.525 1.525 0 1 1-3.05 0 1.525 1.525 0 0 1 3.05 0Zm.9 0a2.425 2.425 0 1 1-4.85 0 2.425 2.425 0 0 1 4.85 0Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Configuración
              </span>
            </a>
          </li>
          <li>
            <button
              class="logout-button"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 15 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3 1.5a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h7.5a.5.5 0 0 0 0-1H3v-11h7.5a.5.5 0 0 0 0-1H3Zm9.604 3.896a.5.5 0 0 0-.708.708L13.293 7.5H6.5a.5.5 0 0 0 0 1h6.793l-1.397 1.396a.5.5 0 0 0 .708.708l2.25-2.25a.5.5 0 0 0 0-.708l-2.25-2.25Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="logout-button__text"
              >
                Cerrar Sesión
              </span>
            </button>
          </li>
        </ul>
      </nav>
      <footer
        class="sidebar__footer"
      >
        <div
          class="sidebar__profile-image"
        >
          <span>
            J
          </span>
        </div>
        <div
          class="sidebar_profile-information"
        >
          <p
            class="profile__username"
          >
            Juan Pérez
          </p>
          <p
            class="profile__email"
          >
            d@d.com
          </p>
        </div>
      </footer>
    </aside>
    <main
      class="crop-container"
    >
      <header
        class="crop__header"
      >
        <a
          class="back crop__backButton"
          data-discover="true"
          href="/home"
        >
          <svg
            fill="#065e52"
            height="30"
            viewBox="0 0 15 15"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <h1>
          CULTIVOS
        </h1>
      </header>
      <section
        class="crop__content"
      >
        <div
          class="crop__content__actions"
        >
          <button
            class="btnAdd"
          >
            <svg
              fill="#008164"
              height="30"
              viewBox="0 0 15 15"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.49991 0.876892C3.84222 0.876892 0.877075 3.84204 0.877075 7.49972C0.877075 11.1574 3.84222 14.1226 7.49991 14.1226C11.1576 14.1226 14.1227 11.1574 14.1227 7.49972C14.1227 3.84204 11.1576 0.876892 7.49991 0.876892ZM1.82707 7.49972C1.82707 4.36671 4.36689 1.82689 7.49991 1.82689C10.6329 1.82689 13.1727 4.36671 13.1727 7.49972C13.1727 10.6327 10.6329 13.1726 7.49991 13.1726C4.36689 13.1726 1.82707 10.6327 1.82707 7.49972ZM7.50003 4C7.77617 4 8.00003 4.22386 8.00003 4.5V7H10.5C10.7762 7 11 7.22386 11 7.5C11 7.77614 10.7762 8 10.5 8H8.00003V10.5C8.00003 10.7761 7.77617 11 7.50003 11C7.22389 11 7.00003 10.7761 7.00003 10.5V8H4.50003C4.22389 8 4.00003 7.77614 4.00003 7.5C4.00003 7.22386 4.22389 7 4.50003 7H7.00003V4.5C7.00003 4.22386 7.22389 4 7.50003 4Z"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-fkSzgi kmvUTK"
        >
          <div
            class="sc-hFflus fAWkUP"
          >
            <div
              class="sc-KXCwU dAHBzl rdt_Table"
              role="table"
            >
              <div
                class="sc-gBoluC ewUEli"
              >
                <div
                  style="padding: 24px;"
                >
                  There are no records to display
                </div>
              </div>
            </div>
          </div>
        </div>
        <dialog
          class="crop-modal"
        >
          <button
            class="close-button crop-modal__btn-close"
          >
            <svg
              fill="#8b8d98"
              height="20"
              viewBox="0 0 15 15"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <section
            class="crop-modal__container"
          >
            <h2>
              Añadir Cultivo
            </h2>
            <form
              class="crop-modal__form"
            >
              <div
                class="crop-modal__form__inputs"
              >
                <label
                  class="crop-modal__form__label"
                  for="cropName"
                >
                  Nombre del Cultivo
                </label>
                <input
                  class="crop-modal__form__input"
                  id="cropName"
                  name="nombre"
                  placeholder="Ingrese el nombre del cultivo"
                  required=""
                  type="text"
                />
                <label
                  class="crop-modal__form__label"
                  for="typeCrop"
                >
                  Tipo de cultivo
                </label>
                <input
                  class="crop-modal__form__input"
                  id="typeCrop"
                  name="tipo_siembra"
                  placeholder="ej( Café, Banano, Frijol. etc)"
                  required=""
                  type="text"
                />
                <label
                  class="crop-modal__form__label"
                  for="dateCrop"
                >
                  Fecha de inicio
                </label>
                <input
                  class="crop-modal__form__input"
                  id="dateCrop"
                  name="fecha_inicio"
                  required=""
                  type="date"
                />
                <div
                  class="crop-modal__form__inputs--area"
                >
                  <label
                    class="crop-modal__form__label area-label"
                    for="areaCrop"
                  >
                    Area del terreno
                  </label>
                  <input
                    class="crop-modal__form__input"
                    id="areaCrop"
                    name="area_terreno"
                    placeholder="0"
                    required=""
                    type="number"
                  />
                  <select
                    class="crop-modal__form__units"
                    id="unit"
                    name="unidad"
                    required=""
                    title="Selecciona la unidad de medida"
                  >
                    <option
                      disabled=""
                      selected=""
                      value=""
                    >
                      Elija una unidad
                    </option>
                  </select>
                </div>
              </div>
              <div
                class="crop-modal__form__buttons"
              >
                <button
                  class="primary-button__component undefined"
                  type="submit"
                >
                  Añadir Cultivo
                </button>
                <button
                  class="secondary-button"
                  type="button"
                >
                  Cancelar
                </button>
              </div>
            </form>
          </section>
        </dialog>
      </section>
    </main>
  </div>
</div>
`;
